src/vivarium_public_health/dataset_manager/artifact.py

```diff
@@ -31,11 +37,11 @@ def __init__(self, path: str, filter_terms: List[str]=None):
             A set of terms suitable for usage with the ``where`` kwarg for ``pd.read_hdf``
         """
 
+        create_hdf_with_keyspace(path)
```

Comment:
so if someone creates an Artifact and gives it a path that doesn't exist, this will now create an empty Artifact? do we want to be doing that here?

Reply:
that's what we concluded after we had an issue with initializing artifact using the `meatdata.keyspace`, if I understand correctly. It is related to with a comment from James last week regarding the encapsulation issue

Reply:
hmm now I'm thinking about this more. Does this mean that if I accidentally mistype the artifact path in my yaml config and then try to run a sim, I'm going to end up creating an empty Artifact and then getting an error that the first key I try to read isn't in the artifact rather than the FileNotFoundError I would previously get? Not a big deal but we might want to figure out how to catch it so the error's less confusing at some point

Reply:
yeah, I agree, catching error and giving a proper message are really important. But it's pretty difficult to tell from the beginning whether the artifact has proper data or just newly made empty one :( For that, we probably need a bit bigger change around the artifact, like including the list of components or so, since keyspace doesn't really tell you whether this artifact is enough for to run your configuration